nixos/modules/system/boot/stage-1-init.sh

```diff
@@ -423,8 +423,15 @@ mountFS() {
 
     # For bind mounts, busybox has a tendency to ignore options, which can be a
     # security issue (e.g. "nosuid"). Remounting the partition seems to fix the
-    # issue.
-    mount "/mnt-root$mountPoint" -o "remount,$optionsPrefixed"
+    # issue. This should only be done for bind and rbind mountpoints because other
+    # filesystems may have limitations or may not support remount.
+    local IFS=,
+    for opt in $options; do
+        if [ "$opt" = bind ] || [ "$opt" = rbind ]; then
```

Comment:
Why not this?

```suggestion
        if [[ "$opt" = bind || "$opt" = rbind ]]; then
```

Reply:
I just wanted to use the same syntax as at [line 133] (https://github.com/NixOS/nixpkgs/blob/a7cbfb346d5ca0f1c96d535339bda3a0a192b265/nixos/modules/system/boot/stage-1-init.sh#L133) just to be consistent but I've updated the PR to use the more preferred syntax. (Unfortunately stage-1-init.sh has a lot of issues according the shellcheck.)